apps/lend/src/components/AlertLoanSummary/components/SummaryFull.tsx

```diff
@@ -12,33 +12,38 @@ const SummaryFull = ({
   pendingMessage,
   receive = '',
   formValueStateCollateral = '',
-  formValueUserBorrowed = '',
   userState,
   collateralSymbol,
   borrowedSymbol,
 }: SummaryProps) => {
-  const { debt: stateDebt = '', collateral: stateCollateral = '' } = userState ?? {}
-
-  const returnToWallet = useMemo(() => {
-    return [
-      {
-        ...(+receive > 0
-          ? { value: +receive - +stateDebt, symbol: borrowedSymbol }
-          : { value: +formValueUserBorrowed - +stateDebt, symbol: borrowedSymbol }),
-      },
-      { value: +stateCollateral - +formValueStateCollateral, symbol: collateralSymbol },
-    ].filter(({ value }) => +value > 0)
-  }, [
-    borrowedSymbol,
-    collateralSymbol,
-    formValueStateCollateral,
-    formValueUserBorrowed,
-    receive,
-    stateCollateral,
-    stateDebt,
-  ])
-
-  const balance = +stateDebt - +receive - +formValueUserBorrowed
+  const { debt: stateDebt = '', collateral: stateCollateral = '', borrowed: stateBorrowed = '' } = userState ?? {}
+
+  const { balance, returnToWallet } = useMemo(() => {
+    let resp = { balance: 0, returnToWallet: [] as { value: number | string; symbol: string }[] }
```

Comment:
For the TS compiler to work well and to check type matching, it is best to use `as` for type conversion as rarely as possible. It is better to specify the type globally (like the return type of useMemo) and TS will automatically keep an eye on the variable to make sure it conforms to the type constraint.